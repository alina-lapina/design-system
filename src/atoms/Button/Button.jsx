import React from 'react';
import PropTypes from 'prop-types';

const Button = ({
	danger, disabled, icon, onClick, primary, text, type,
}) => (
	<button
		type={type}
		className={
			`button-container ${primary ? 'primary' : 'secondary'} ${danger && 'danger'}`
		}
		onClick={onClick}
		disabled={disabled}
	>	{icon}
		<span className="button-text">{text}</span>
	</button>
);

Button.defaultProps = {
	danger: false,
	disabled: false,
	primary: false,
	type: 'button',
};

Button.propTypes = {
	danger: PropTypes.bool,
	disabled: PropTypes.bool,
	icon: PropTypes.element,
	onClick: PropTypes.func,
	primary: PropTypes.bool,
	text: PropTypes.string.isRequired,
	type: PropTypes.string,
};

export default Button;
