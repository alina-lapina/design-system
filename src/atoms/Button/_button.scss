$regular-disable-color: #a2baba;
$negative-disable-color: $label-color;

.button-container {
  align-items: center;
  border-radius: 2px;
  display: flex;
  height: fit-content;
  margin-right: 20px;
  padding: $global-padding-size ($global-padding-size * 2);

  &:hover, &:active {
    cursor: pointer;
  }

  &:disabled {
    cursor: not-allowed;
  }
}

.secondary {
  background: $ssb-white;
  border: 1px solid $ssb-green-4;
  color: $ssb-green-4;

  &:hover, &:active {
    color: $ssb-white;

    .button-text {
      border-bottom: 1px solid $ssb-white;
    }
  }

  &:hover {background: $ssb-green-4;}

  &:active {background: $ssb-green-5;}

  &:disabled {
    background: $ssb-white;
    color: $regular-disable-color;
    border: 1px solid $regular-disable-color;

    .button-text {
      border-bottom: none;
    }
  }
}

.primary {
  background: $ssb-green-4;
  color: $ssb-white;

  &:hover {
    background: $ssb-green-5;

    .button-text {
      border-bottom: 1px solid $ssb-white;
    }
  }

  &:disabled {
    background: $regular-disable-color;
    border: none;

    .button-text {
      border-bottom: none;
    }
  }
}
