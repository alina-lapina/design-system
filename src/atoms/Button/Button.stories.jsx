import React from 'react';
import { storiesOf } from '@storybook/react';
import centered from '@storybook/addon-centered/react';
import Button from './Button';

storiesOf('Button', module).addDecorator(centered)
	.add('Primary', () => (
		<div>
			<Button text="Primary" primary />
			<Button text="Disabled" primary disabled />
		</div>
	))
	.add('Secondary', () => (
		<div>
			<Button text="Secondary" />
			<Button text="Disabled" disabled />
		</div>
	))
	.add('Danger', () => (
		<div>
			<Button text="Danger" primary danger />
		</div>
	));
