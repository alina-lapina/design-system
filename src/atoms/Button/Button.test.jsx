import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

describe('Button component', () => {
	test('Matches the snapshot', () => {
		const button = shallow(<Button text="Button" />);
		expect(button).toMatchSnapshot();
	});

	test('Adds class if "danger" is true', () => {
		const button = shallow(<Button text="Button" danger />);
		expect(button.find('button').hasClass('danger')).toEqual(true);
	});

	test('Changes class if it is a primary button', () => {
		const button = shallow(<Button text="Button" primary />);
		expect(button.find('button').hasClass('primary')).toEqual(true);
	});
});
