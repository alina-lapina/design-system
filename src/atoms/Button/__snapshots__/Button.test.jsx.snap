// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Matches the snapshot 1`] = `
<button
  className="container secondary false"
  disabled={false}
  type="button"
>
   
  <span
    className="button-text"
  >
    Button
  </span>
</button>
`;
